docs/reference-guide/query-modules/implement-custom-query-modules/api/python-api.md

```diff
@@ -139,6 +139,26 @@ CALL example.procedure(“single argument”) YIELD result;
 Naturally, you may pass in different arguments.
 
 
+## mgp.add_batch_read_proc(func: Callable[…, mgp.Record], initializer: typing.Callable, cleanup: typing.Callable)
+
+Register func as a read-only batch procedure of the current module.
+
+`func` represents a function that is invoked through OpenCypher. Through OpenCypher user invokes `func`. Memgraph invokes first the `initializer` function. After the `initializer` function, `func` is called until returns an empty result. Afterward, the `cleanup` function will be called, which can be used to clean up global resources. Only at that point is garbage collection invoked, so any dangling references to Python objects will be cleaned.
+
+`initializer` must define the same parameters as the main `func` function, and will receive the same parameters as `func`. The position of arguments and the type of arguments must be the same. 
+
+
+Otherwise, the same rules apply as in `read_proc`. Important to keep in mind is that any Memgraph resources can't be stored in `init` and during batching. All objects are after each call invalidated and referencing those will result in error.
+
+One more thing to keep in mind is that nodes, relationships or any other Memgraph types can't be stored as global variables. After `initializer` and each `func` call, every Memgraph related object is invalidated and can't be used later on.
```

Comment:
```suggestion
Otherwise, the same rules apply as in `read_proc`. It's important to keep in mind that no Memgraph resources can be stored in `init` and during batching. After each call, all objects are invalidated and referencing them will result in an error.

Also, nodes, relationships or any other Memgraph types can't be stored as global variables. After `initializer` and each `func` call, every Memgraph-related object is invalidated and can't be used later on.
```

Reply:
I hope i got this right

Reply:
Not all are invalidated, just Memgraph resources.  So you can still have resources like list of strings, but not list of vertices o relationships.